creator-node/src/routes/tracks.js

```diff
@@ -344,55 +347,61 @@ module.exports = function (app) {
               multihash: trackSegmentCIDs,
               cnodeUserUUID,
               trackBlockchainId: null,
-              type: 'track'
+              type: 'track',
+              sourceFile: transcodedFile.sourceFile
             },
             transaction
           })
 
-          if (trackFiles.length < trackSegmentCIDs.length) {
+          if (trackFiles.length !== trackSegmentCIDs.length) {
             req.logger.error(
               `Did not find files for every track segment CID for user ${cnodeUserUUID} ${trackFiles} ${trackSegmentCIDs}`
             )
             throw new Error('Did not find files for every track segment CID.')
           }
-          const numAffectedRows = await models.File.update(
+          const segmentsAssociateNumAffectedRows = await models.File.update(
             { trackBlockchainId: track.blockchainId },
             {
               where: {
                 multihash: trackSegmentCIDs,
                 cnodeUserUUID,
                 trackBlockchainId: null,
-                type: 'track'
+                type: 'track',
+                sourceFile: transcodedFile.sourceFile
               },
               transaction
             }
           )
-          if (parseInt(numAffectedRows, 10) < trackSegmentCIDs.length) {
+          if (
```

Comment:
since the trackBlockchainId is null here, what if you upload multiple copies of a previously uploaded track, would it associate correctly?

Reply:
it now uniques on sourceFile so as long as you pass in transcodedTrackUUID it should ignore all the files from previously uploaded tracks right

Reply:
just confirming this makes sense to you @dmanjunath